Added a highscore Manager

MV1090 opened the pull request:
I created a high score manager from a singleton, this way the high score doesn't reset when the game is started from the title screen. I moved all the high score code from the game manager, the canvas manager now calls the getHighScore()method from HSM instead of GM. everything appears to work, I'm not sure if this was the best way to approach this so let me know if you have a different idea.

Commit:
Added a highscore Manager

I created a high score manager from a singlton, this way the high score doesn't reset, when the game is started from the title screen. I moved all the high score code from the game manager, the canvas manager now calls the getHighScore()method from HSM instead of GM. everything works, I'm not sure if this was the best way to approach this so let me know if you have a different idea.

SolomanDavis commented on Assets/_Scripts/Managers/HighScoreManager.cs:
```diff
@@ -0,0 +1,27 @@
+        if (_score > _highScore)
+        {
+            _highScore = _score;
+        }
```
It's dangerous / misleading for a `GetHighScore` method to also set the high score.

This should be split into two separate methods: a getter and a setter.

The `GetHighScore` should be called by the `CanvasManager` to display the correct text and the `SetHighScore` should be called by the `GameManager` on win to set the high score if applicable.

MV1090 replied:
Yeah good point. Set high score is now called in a GameManager both when the player dies and wins.

Commit:
Split method in to Get and Set.

Split the highScore method in to a Getter and Setter, The canvas manager Gets the high score and prints it to screen. The GameManager sets the High score when the player dies or wins, only if their current score is bigger then the highscore.
